jacodb-panda-dynamic/src/main/kotlin/org/jacodb/panda/dynamic/ets/base/EtsExpr.kt

```diff
@@ -0,0 +1,309 @@
+
+            fun defaultVisit(expr: EtsExpr): R
+        }
+    }
+
+    override fun <R> accept(visitor: EtsEntity.Visitor<R>): R {
+        return accept(visitor as Visitor<R>)
+    }
+
+    fun <R> accept(visitor: Visitor<R>): R
+}
+
+data class EtsNewExpr(
+    override val type: EtsType, // ClassType
+) : EtsExpr {
+    override fun toString(): String {
+        return "new ${type.typeName}"
+    }
+
+    override fun <R> accept(visitor: EtsExpr.Visitor<R>): R {
+        return visitor.visit(this)
+    }
+}
+
+data class EtsDeleteExpr(
+    val arg: EtsEntity,
+) : EtsExpr {
+    override val type: EtsType
+        get() = EtsBooleanType
+
+    override fun toString(): String {
+        return "delete $arg"
+    }
+
+    override fun <R> accept(visitor: EtsExpr.Visitor<R>): R {
+        return visitor.visit(this)
+    }
+}
+
+data class EtsNewArrayExpr(
+    val elementType: EtsType,
+    val size: EtsEntity,
+) : EtsExpr {
+    // TODO: support multi-dimensional arrays
+    override val type: EtsType
+        get() = EtsArrayType(elementType, 1)
+
+    override fun toString(): String {
+        return "new ${elementType.typeName}[$size]"
+    }
+
+    override fun <R> accept(visitor: EtsExpr.Visitor<R>): R {
+        return visitor.visit(this)
+    }
+}
+
+data class EtsTypeOfExpr(
+    val arg: EtsEntity,
+) : EtsExpr {
+    override val type: EtsType
+        get() = EtsStringType
+
+    override fun toString(): String {
+        return "typeof $arg"
+    }
+
+    override fun <R> accept(visitor: EtsExpr.Visitor<R>): R {
+        return visitor.visit(this)
+    }
+}
+
+data class EtsInstanceOfExpr(
+    val arg: EtsEntity,
+    val checkType: EtsType,
+) : EtsExpr {
+    override val type: EtsType
+        get() = EtsBooleanType
+
+    override fun toString(): String {
+        return "$arg instanceof $checkType"
+    }
+
+    override fun <R> accept(visitor: EtsExpr.Visitor<R>): R {
+        return visitor.visit(this)
+    }
+}
+
+data class EtsLengthExpr(
+    val arg: EtsEntity,
+) : EtsExpr {
+    override val type: EtsType
+        get() = EtsNumberType
+
+    override fun toString(): String {
+        return "$arg.length"
+    }
+
+    override fun <R> accept(visitor: EtsExpr.Visitor<R>): R {
+        return visitor.visit(this)
+    }
+}
+
+data class EtsCastExpr(
+    val arg: EtsEntity,
+    override val type: EtsType,
+) : EtsExpr {
+    override fun toString(): String {
+        return "$arg as $type"
+    }
+
+    override fun <R> accept(visitor: EtsExpr.Visitor<R>): R {
+        return visitor.visit(this)
+    }
+}
+
+// data class TernaryExpr(
+//     val condition: Value,
+//     val trueBranch: Value,
+//     val falseBranch: Value,
+// ) : Expr {
+//     override val type: Type
+//         get() = TypeInference.commonType(trueBranch.type, trueBranch.type)
+//
+//     override fun toString(): String {
+//         return "$condition ? $trueBranch : $falseBranch"
+//     }
+// }
```

Comment:
There is no TernaryExpr in ArkIR, and initially, when I designed "a better version of ArkIR, in Kotlin", I though it would be nice to have it. I think we can safely drop it.